modules/vstudio/vs2010_vcxproj.lua

```diff
@@ -3501,14 +3805,14 @@
 	function m.linuxApplicationType(cfg)
 		m.element("Keyword", nil, "Linux")
 		m.element("RootNamespace", nil, "%s", cfg.project.name)
-		m.element("MinimumVisualStudioVersion", nil, "17.0")
+		m.element("MinimumVisualStudioVersion", nil, "15.0")
```

Comment:
I thought the MinimumVisualStudioVersion was there to prevent opening a project in earlier versions. The reason I changed it is that I was testing Linux on VS2019 and it wasn't letting me open, so I went to VS2017 to see if it let me create the project, which it did. I can do a mapping from action to number no problem. Something like

vs2017 -> 15
vs2019 -> 16
vs2022 -> 17

Not sure what you mean by "so setting this below the current action version is probably wrong". If I generate a vs2022 Linux project and the minimum is 15, that means it all works fine. If I open it in VS2015 instead, the project won't load. That means you'd have to have a check in premake.lua to exclude it from VS2015

Reply:
The mapping already exists in each VS `newaction` call: https://github.com/premake/premake-core/blob/8f174c409ab88b3c744441fb7e15085e9c014678/modules/vstudio/vs2015.lua#L65

Which we use like this: https://github.com/premake/premake-core/blob/8f174c409ab88b3c744441fb7e15085e9c014678/modules/vstudio/vs2010_vcxproj.lua#L66-L74

> Not sure what you mean by "so setting this below the current action version is probably wrong".

I don't have a specific example to provide, but this kind of thing does exist in the normal VS path. When VS introduces new settings, old versions end up ignoring it. With the VS Linux implementation in particular, they have made improvements since their vs2015 extension, including a number of new settings which old versions won't understand. So, while an older VS _could_ open it, it may not work due to those missing settings - or may work in a broken or unexpected way.